Installing `gd` and `vips` extensions causes a build error

### Version of install-php-extensions

latest

### Error description

Hi,

I'm currently trying to build a local dev docker image to run an app locally.

Not sure if this is the right project to put this, if not I'd be happy if you could guide me to the right one.

I get the following errors during the `RUN install-php-extensions ...` build step, it's related to CMake:
**Note**: It's just tail of the log, but basically it's a lot of errors related to libavif

```
42.31 /tmp/src/tmp.tA1D6t988N/libavif-1.0.3/src/codec_aom.c:519:17: error: 'AOME_THREEFOUR' undeclared here (not in a function); did you mean 'AOME_THREEFIVE'?
42.31   519 |     { { 3, 4 }, AOME_THREEFOUR }, { { 3, 5 }, AOME_THREEFIVE }, { { 4, 5 }, AOME_FOURFIVE },
42.31       |                 ^~~~~~~~~~~~~~
42.31       |                 AOME_THREEFIVE
42.31 /tmp/src/tmp.tA1D6t988N/libavif-1.0.3/src/codec_aom.c:519:17: error: incompatible types when initializing type 'enum aom_scaling_mode_1d' using type 'const struct aomOptionDef *'
42.31 /tmp/src/tmp.tA1D6t988N/libavif-1.0.3/src/codec_aom.c: In function 'aomCodecEncodeImage':
42.31 /tmp/src/tmp.tA1D6t988N/libavif-1.0.3/src/codec_aom.c:580:33: error: 'AOM_USAGE_GOOD_QUALITY' undeclared (first use in this function)
42.31   580 |         unsigned int aomUsage = AOM_USAGE_GOOD_QUALITY;
42.31       |                                 ^~~~~~~~~~~~~~~~~~~~~~
42.31 /tmp/src/tmp.tA1D6t988N/libavif-1.0.3/src/codec_aom.c:580:33: error: initialization of 'unsigned int' from 'const struct aomScalingModeMapList *' makes integer from pointer without a cast [-Werror=int-conversion]
42.31 /tmp/src/tmp.tA1D6t988N/libavif-1.0.3/src/codec_aom.c:597:28: error: 'AOM_USAGE_REALTIME' undeclared (first use in this function)
42.31   597 |                 aomUsage = AOM_USAGE_REALTIME;
42.31       |                            ^~~~~~~~~~~~~~~~~~
42.31 /tmp/src/tmp.tA1D6t988N/libavif-1.0.3/src/codec_aom.c:597:26: error: assignment to 'unsigned int' from 'const struct aomScalingModeMapList *' makes integer from pointer without a cast [-Werror=int-conversion]
42.31   597 |                 aomUsage = AOM_USAGE_REALTIME;
42.31       |                          ^
42.31 cc1: all warnings being treated as errors
42.31 make[2]: *** [CMakeFiles/avif.dir/build.make:303: CMakeFiles/avif.dir/src/codec_aom.c.o] Error 1
42.31 make[2]: *** Waiting for unfinished jobs....
42.53 make[1]: *** [CMakeFiles/Makefile2:135: CMakeFiles/avif_internal.dir/all] Error 2
42.53 make[1]: *** Waiting for unfinished jobs....
42.57 make[1]: *** [CMakeFiles/Makefile2:162: CMakeFiles/avif.dir/all] Error 2
42.57 make: *** [Makefile:149: all] Error 2
------
Dockerfile:9
--------------------
   7 |     ADD --chmod=0755 https://github.com/mlocati/docker-php-extension-installer/releases/latest/download/install-php-extensions /usr/local/bin/
   8 |     
   9 | >>> RUN install-php-extensions pdo_mysql mysqli zip xsl gd intl opcache exif ffi vips
  10 |     
  11 |     RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
--------------------
ERROR: failed to solve: process "/bin/sh -c install-php-extensions pdo_mysql mysqli zip xsl gd intl opcache exif ffi vips" did not complete successfully: exit code: 2
```

Removing `vips` from the dependencies will fix it, so it's definetly related to that.

**Also note**: I've tried php versions 8.1, 8.2, and 8.3. They all produce the error.

### Docker image

php:8.3-fpm-bullseye

### Minimal Dockerfile

```Dockerfile
FROM php:8.3-fpm-bullseye

RUN apt-get update && apt-get install -y \
        git \
        unzip 

ADD --chmod=0755 https://github.com/mlocati/docker-php-extension-installer/releases/latest/download/install-php-extensions /usr/local/bin/

RUN install-php-extensions pdo_mysql mysqli zip xsl gd intl opcache exif ffi vips

RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
```

--- a/install-php-extensions
+++ b/install-php-extensions
@@ -662,9 +662,11 @@ expandASpellDictionaries() {
 #   PACKAGES_PERSISTENT_PRE the list of packages required at runtume that are already installed
 #   PACKAGES_VOLATILE the list of packages required at compile time that must be installed
 #   PACKAGES_PREVIOUS the list of packages (with their version) that are installed right now (calculated only on Debian and only if PACKAGES_PERSISTENT_NEW or PACKAGES_VOLATILE are not empty)
+#   COMPILE_LIBS
 buildRequiredPackageLists() {
 	buildRequiredPackageLists_persistent=''
 	buildRequiredPackageLists_volatile=''
+	COMPILE_LIBS=''
 	case "$DISTRO" in
 		alpine)
 			apk update
@@ -844,6 +846,18 @@ buildRequiredPackageLists() {
 							case "${IPE_GD_WITHOUTAVIF:-}" in
 								1 | y* | Y*) ;;
 								*)
+									if ! isLibaomInstalled; then
+										COMPILE_LIBS="$COMPILE_LIBS libaom"
+									fi
+									if ! isLibdav1dInstalled; then
+										COMPILE_LIBS="$COMPILE_LIBS libdav1d"
+									fi
+									if ! isLibyuvInstalled; then
+										COMPILE_LIBS="$COMPILE_LIBS libyuv"
+									fi
+									if ! isLibavifInstalled; then
+										COMPILE_LIBS="$COMPILE_LIBS libavif"
+									fi
 									buildRequiredPackageLists_persistent="$buildRequiredPackageLists_persistent libstdc++"
 									buildRequiredPackageLists_volatile="$buildRequiredPackageLists_volatile cmake nasm meson"
 									;;
@@ -869,6 +883,18 @@ buildRequiredPackageLists() {
 							case "${IPE_GD_WITHOUTAVIF:-}" in
 								1 | y* | Y*) ;;
 								*)
+									if ! isLibaomInstalled; then
+										COMPILE_LIBS="$COMPILE_LIBS libaom"
+									fi
+									if ! isLibdav1dInstalled; then
+										COMPILE_LIBS="$COMPILE_LIBS libdav1d"
+									fi
+									if ! isLibyuvInstalled; then
+										COMPILE_LIBS="$COMPILE_LIBS libyuv"
+									fi
+									if ! isLibavifInstalled; then
+										COMPILE_LIBS="$COMPILE_LIBS libavif"
+									fi
 									buildRequiredPackageLists_volatile="$buildRequiredPackageLists_volatile cmake nasm meson"
 									;;
 							esac
@@ -2472,6 +2498,22 @@ You may need to:
 EOT
 }
 
+# Compile the libraries specified in the COMPILE_LIBS variable
+compileLibs() {
+	if stringInList libaom "$COMPILE_LIBS"; then
+		installLibaom
+	fi
+	if stringInList libdav1d "$COMPILE_LIBS"; then
+		installLibdav1d
+	fi
+	if stringInList libyuv "$COMPILE_LIBS"; then
+		installLibyuv
+	fi
+	if stringInList libavif "$COMPILE_LIBS"; then
+		installLibavif
+	fi
+}
+
 # Install a bundled PHP module given its handle
 #
 # Arguments:
@@ -2568,23 +2610,6 @@ EOF
 						;;
 				esac
 				if test $installBundledModule_tmp -eq 0; then
-					case "${IPE_GD_WITHOUTAVIF:-}" in
-						1 | y* | Y*) ;;
-						*)
-							if ! isLibaomInstalled; then
-								installLibaom
-							fi
-							if ! isLibdav1dInstalled; then
-								installLibdav1d
-							fi
-							if ! isLibyuvInstalled; then
-								installLibyuv
-							fi
-							if ! isLibavifInstalled; then
-								installLibavif
-							fi
-							;;
-					esac
 					if isLibaomInstalled && isLibdav1dInstalled && isLibyuvInstalled && isLibavifInstalled; then
 						installBundledModule_tmp=1
 					fi
@@ -4828,6 +4853,8 @@ fi
 if test $USE_PICKLE -gt 1; then
 	buildPickle
 fi
+
+compileLibs
 
 for PHP_MODULE_TO_INSTALL in $PHP_MODULES_TO_INSTALL; do
 	case "$PHP_MODULE_TO_INSTALL" in

--- a/scripts/ci-test-extensions
+++ b/scripts/ci-test-extensions
@@ -127,13 +127,20 @@ filterUnsupportedExensionsForDistro() {
 			;;
 	esac
 
-	for filterUnsupportedExensionsForDistro_extension in $EXTENSIONS_TO_BE_TESTED; do
-		if stringInList "!$IPETEST_DOCKER_DISTRO" "$(cat "$filterUnsupportedExensionsForDistro_reqs" | grep -E "^$filterUnsupportedExensionsForDistro_extension[ \t]")"; then
-			printf 'Note: extension "%s" is not supported for distro "%s"\n' "$filterUnsupportedExensionsForDistro_extension" "$IPETEST_DOCKER_DISTRO"
-		elif stringInList "!$filterUnsupportedExensionsForDistro_baseDistro" "$(cat "$filterUnsupportedExensionsForDistro_reqs" | grep -E "^$filterUnsupportedExensionsForDistro_extension[ \t]")"; then
-			printf 'Note: extension "%s" is not supported for distro "%s"\n' "$filterUnsupportedExensionsForDistro_extension" "$filterUnsupportedExensionsForDistro_baseDistro"
-		else
-			filterUnsupportedExensionsForDistro_filtered="$filterUnsupportedExensionsForDistro_filtered $filterUnsupportedExensionsForDistro_extension"
+	for filterUnsupportedExensionsForDistro_extensions in $EXTENSIONS_TO_BE_TESTED; do
+		filterUnsupportedExensionsForDistro_ok=1
+		IFS='+'
+		for filterUnsupportedExensionsForDistro_extension in $filterUnsupportedExensionsForDistro_extensions; do
+			if stringInList "!$IPETEST_DOCKER_DISTRO" "$(cat "$filterUnsupportedExensionsForDistro_reqs" | grep -E "^$filterUnsupportedExensionsForDistro_extension[ \t]")"; then
+				printf 'Note: extension "%s" is not supported for distro "%s"\n' "$filterUnsupportedExensionsForDistro_extension" "$IPETEST_DOCKER_DISTRO"
+				filterUnsupportedExensionsForDistro_ok=0
+			elif stringInList "!$filterUnsupportedExensionsForDistro_baseDistro" "$(cat "$filterUnsupportedExensionsForDistro_reqs" | grep -E "^$filterUnsupportedExensionsForDistro_extension[ \t]")"; then
+				printf 'Note: extension "%s" is not supported for distro "%s"\n' "$filterUnsupportedExensionsForDistro_extension" "$filterUnsupportedExensionsForDistro_baseDistro"
+				filterUnsupportedExensionsForDistro_ok=0
+			fi
+		done
+		if test $filterUnsupportedExensionsForDistro_ok -eq 1; then
+			filterUnsupportedExensionsForDistro_filtered="$filterUnsupportedExensionsForDistro_filtered $filterUnsupportedExensionsForDistro_extensions"
 		fi
 	done
 	resetIFS
@@ -164,10 +171,13 @@ getDockerImageName() {
 	getDockerImageName_reqs="$CI_BUILD_DIR/data/special-requirements"
 	if test -f "$getDockerImageName_reqs"; then
 		IFS=' '
-		for getDockerImageName_testExtension in $1; do
-			if test -n "$(cat "$getDockerImageName_reqs" | grep -E "^$getDockerImageName_testExtension[ \t]+zts[ \t]*$")"; then
-				getDockerImageName_suffix='zts'
-			fi
+		for getDockerImageName_testExtensions in $1; do
+			IFS='+'
+			for getDockerImageName_testExtension in $getDockerImageName_testExtensions; do
+				if test -n "$(cat "$getDockerImageName_reqs" | grep -E "^$getDockerImageName_testExtension[ \t]+zts[ \t]*$")"; then
+					getDockerImageName_suffix='zts'
+				fi
+			done
 		done
 	fi
 	getDockerImageName_imageName="$(printf 'php:%s-%s-%s' "$getDockerImageName_version" "$getDockerImageName_suffix" "$IPETEST_DOCKER_DISTRO")"

